### ISSUE TITLE
Filetype-dependent settings need re-examined

### ISSUE THREAD

**Issue by benknoble** [benknoble/Dotfiles#47]
**This is a lengthy one.**

It comes down to about 3 things:

1. Tab settings are handled ok-ish by the default plugins. Remove 'global' settings. I'd really like to use `expandtab`, obviously. Consider `shiftround` as well. And don't forget `shiftwidth` (`tabstop=8` always)
2. Filetype autocommands can actually be moved to `/.vim/ftplugin/<type>.vim` to set `local` options and use `<localleader>` for mappings.
3. Once 'global' settings are removed, keep an eye out for changes that need to be made and add them.

Plugin files are used for defining file-dependent settings.

Basically, what I want to do is eschew settings that are too global for different filetypes, move my personal prefs into filetype plugins, and then rely on the defaults until I decide I need to change something.

⚠️ 💥 **This will likely require a `:retab` of *all* files!** ❗️❗️

**Comment by benknoble** [comment]
Turns out, setting e.g. `shiftwidth` or `softtabstop` *does* get overridden by ftplugin entries
